src/sentry/identity/oauth2.py

```diff
@@ -102,6 +119,48 @@ def get_oauth_data(self, payload):
 
         return data
 
+    def refresh_identity(self, auth_identity):
+        refresh_token = auth_identity.data.get('refresh_token')
+
+        if not refresh_token:
+            raise IdentityNotValid('Missing refresh token')
+
+        data = self.get_refresh_token_params(
+            refresh_token=refresh_token,
+        )
+
+        req = safe_urlopen(
+            url=self.get_refresh_token_url(),
+            headers=self.get_refresh_token_headers(),
+            data=data,
+        )
+
+        try:
+            body = safe_urlread(req)
+            payload = json.loads(body)
+        except Exception:
+            payload = {}
+
+        error = payload.get('error', 'unknown_error')
+        error_description = payload.get('error_description', 'no description available')
+
+        formatted_error = 'HTTP {} ({}): {}'.format(req.status_code, error, error_description)
+
+        if req.status_code == 401:
+            raise IdentityNotValid(formatted_error)
+
+        if req.status_code == 400:
+            # this may not be common, but at the very least Google will return
+            # an invalid grant when a user is suspended
+            if error == 'invalid_grant':
+                raise IdentityNotValid(formatted_error)
+
+        if req.status_code != 200:
+            raise Exception(formatted_error)
```

Comment:
The error handling is the same code from the old oauth code. I was wondering if that was still the desired behavior here?

Reply:
Yeah this is fine I think, we probably want to know what the error is